Jimmy/content update (#156)
JoshOrndorff/recipes: text/2-appetizers/1-hello-substrate.md

As you can see, our `hello-substrate` pallet has one dipatchable call that takes a single argument, called `origin` which we'll investigate shortly. Both calls return a [`DispatchResult`](https://substrate.dev/rustdocs/master/frame_support/dispatch/type.DispatchResult.html) which can be either `Ok(())` indicating that the call succeeded, or and `Err` which we'll investigate in the [appetizer about errors](./3-errors.md).

## Inside a Dispatchable Call

Let's take a closer look at our dispatchable call.

```rust, ignore
pub fn say_hello(origin) -> DispatchResult {
	// Ensure that the caller is a regular keypair account
	let caller = ensure_signed(origin)?;

	// Print a message
	print("Hello World");
	// Inspecting variables
	debug::info!("Request sent by: {:?}", caller);

	// Indicate that this call succeeded
	Ok(())
}
```

> This function essentially does three things. First, it uses the [`ensure_signed` function](https://substrate.dev/rustdocs/master/frame_system/fn.ensure_signed.html) to ensure that the caller of the function was a regular user who owns a private key. This macro also returns who that caller was, but in this case we don't care who the caller was. In future recipes we'll explore origins other than signed.

Reviewer: ```suggestion
This function essentially does three things. First, it uses the [`ensure_signed` function](https://substrate.dev/rustdocs/master/frame_system/fn.ensure_signed.html) to ensure that the caller of the function was a regular user who owns a private key. This macro also returns who that caller was. We store the caller's identity in the `caller` variable.
```

> Second, it prints a message and log the caller. Notice that we aren't using rust's normal `println!` macro, but rather a special `print` function and `debug::info!` macro. The reason for this is explain in the next section.

Reviewer: ```suggestion
Second, it prints a message and logs the caller. Notice that we aren't using Rust's normal `println!` macro, but rather a special [`print` function](https://substrate.dev/rustdocs/master/sp_runtime/fn.print.html) and `[debug::info!` macro](https://substrate.dev/rustdocs/master/frame_support/debug/macro.info.html). The reason for this is explained in the next section.
```
